### PR TITLE
feat: Add access codes to event tickets route

### LINKED ISSUE #115: Add access codes to tickets route(UI only)
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

Add access as sub route of tickets.
screenshot from 2017-06-02 13-43-24

- Located at `tickets/access-codes`

Please have a look at #97, before starting the work on the issue.

### PR DESCRIPTION
**Pull request by sumedh123**
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

Fixes #115 
access-codes
access-codes2

### CODE REVIEW

File: app/templates/events/view/tickets/access.hbs [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{{#if device.isMobile}}
+  <br><br>
+{{/if}}
+<div class="ui {{if device.isMobile 'centered' ''}} grid">
+  <div class="{{if device.isMobile '' 'row'}}">
+    <h1 class="ui header {{if device.isMobile 'sixteen' 'six'}} wide column">Access Codes</h1>
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: Please add translation helper here.

File: app/templates/events/view/tickets/access.hbs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    <div class="ten wide column right aligned">
+      <div class="ui search">
+        <div class="ui icon input">
+          <input class="prompt" type="text" placeholder="Search">
+          <i class="search icon"></i>
+        </div>
+        <div class="results"></div>
+        &nbsp;&nbsp;
+        <span>{{t 'Show'}}</span>
+        <select class="ui dropdown">
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: use ember semantic ui dropdown here
 http://semantic-org.github.io/Semantic-UI-Ember/#/modules/dropdown

File: app/templates/events/view/tickets/access.hbs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          <th>{{t '#'}}</th>
+          <th>{{t 'Access Code'}}</th>
+          <th>{{t 'Access Code URL'}}</th>
+          <th>{{t 'Accessible Ticket'}}</th>
+          <th>{{t 'Validity'}}</th>
+          <th>{{t 'Status'}}</th>
+          <th>{{t 'Action'}}</th>
+        </tr>
+      </thead>
+      <tbody>
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: @sumedh123 Please fill this with dummy data, use orga server for reference.

**Comment by sumedh123** [comment]
@geekyd @CosmicCoder96 Please review.

File: app/templates/events/view/tickets/access.hbs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        <tr>
+          <td>{{t '1'}}</td>
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: Get rid of the translation helper. Do not use it on dummy data. Lines 63-69. Remove tranlsation helpers

File: app/templates/events/view/tickets/access.hbs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          <td>{{t '#386'}}</td>
+          <td>{{t 'https://github.com'}}</td>
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: Get rid of the translation helper. Do not use it on dummy data. Why would you want to translate a URL?

**Comment by sumedh123** [comment]
@CosmicCoder96 
small
medium
large

**Comment by CosmicCoder96** [comment]
@sumedh123 Also please update the name of the PR "Access Code" is not informative enough

**Comment by CosmicCoder96** [comment]
@sumedh123 deployment link?

**Comment by sumedh123** [comment]
@CosmicCoder96 Yeah give me a moment. I am deploying.

**Comment by sumedh123** [comment]
@CosmicCoder96 https://oefrontend2017.herokuapp.com/events/fcff4bc1/tickets/access

**Comment by sumedh123** [comment]
@CosmicCoder96 Is anything needed here?

File: app/templates/events/view/tickets/access.hbs [hunk 6]
```diff
@@ -0,0 +1,74 @@
+{{#if device.isMobile}}
+  <br><br>
+{{/if}}
+<div class="ui {{if device.isMobile 'centered' ''}} grid">
```
Comment (geekyd) [inline app/templates/events/view/tickets/access.hbs]: Please use `{{if device.isMobile 'centered'}}`

**Comment by sumedh123** [comment]
@geekyd

**Comment by sumedh123** [comment]
@geekyd @niranjan94 Please review

File: app/templates/events/view/tickets/access.hbs [hunk 7]
```diff
@@ -0,0 +1,74 @@
+{{#if device.isMobile}}
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: remove this.

File: app/templates/events/view/tickets/access.hbs [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  <br><br>
+{{/if}}
+<div class="ui {{if device.isMobile 'centered'}} grid">
+  <div class="{{if device.isMobile '' 'row'}}">
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: why?

File: app/templates/events/view/tickets/access.hbs [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+      <br><br>
+    {{/if}}
+    <div class="{{if device.isTablet '' 'right floated ten wide column right aligned'}}">
+      <div class="ui search">
+        <div class="ui icon input">
+          <input class="prompt" type="text" placeholder="Search">
+          <i class="search icon"></i>
+        </div>
+        <div class="results"></div>
+        &nbsp;&nbsp;
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: remove this

File: app/templates/events/view/tickets/access.hbs [hunk 10]
```diff
@@ -0,0 +1,74 @@
+{{#if device.isMobile}}
+  <br><br>
+{{/if}}
+<div class="ui {{if device.isMobile 'centered'}} grid">
+  <div class="{{if device.isMobile '' 'row'}}">
+    <h1 class="ui header {{if device.isMobile 'sixteen' 'six'}} wide column">{{t 'Access Codes'}}</h1>
+    <div class="{{if device.isMobile 'sixteen' 'ten'}} wide column right aligned"><button class="ui button primary">Create an access code</button></div>
+  </div>
+  <div class="{{if device.isMobile '' 'row'}}">
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: why?

File: app/templates/events/view/tickets/access.hbs [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    {{/if}}
+    <div class="{{if device.isTablet '' 'right floated ten wide column right aligned'}}">
+      <div class="ui search">
+        <div class="ui icon input">
+          <input class="prompt" type="text" placeholder="Search">
+          <i class="search icon"></i>
+        </div>
+        <div class="results"></div>
+        &nbsp;&nbsp;
+        {{#if device.isMobile}}
```
Comment (CosmicCoder96) [inline app/templates/events/view/tickets/access.hbs]: For now, I don't think there is a need of so much styling please keep the general look and feel similar across the tabs.

Comment (sumedh123) [inline app/templates/events/view/tickets/access.hbs]: Okay

File: app/templates/events/view/tickets/access.hbs (hunk 7 above)
Comment (sumedh123) [inline app/templates/events/view/tickets/access.hbs]: I have pushed to heroku. The link is same as before. Please check UI and tell me if I should make the changes you told

**Comment by sumedh123** [comment]
@CosmicCoder96 I have updated the PR with simple style as you suggested. Please review. Same heroku link

**Comment by sumedh123** [comment]
@CosmicCoder96 @geekyd  Looks good?

File: app/templates/events/view/tickets/access.hbs [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+  <div class="row">
+    <div class="{{if device.isMobile 'eight wide column center' 'left'}} aligned">
+      <div class="ui very basic buttons">
+        <button class="ui button active">{{t 'All'}}</button>
+        <button class="ui button">{{t 'Active'}}</button>
+        <button class="ui button">{{t 'Inactive'}}</button>
+      </div>
+    </div>
+    <div class="five wide column {{if device.isMobile 'center' 'right'}} aligned right floated">
+      <div class="ui search">
```
Comment (niranjan94) [inline app/templates/events/view/tickets/access.hbs]: Like I said in the channel, remove search

File: app/templates/events/view/tickets/access.hbs [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        <div class="ui icon input">
+          <input class="prompt" type="text" placeholder="Search">
+          <i class="search icon"></i>
+        </div>
+        <div class="results"></div>
+      </div>
+    </div>
+    <div class="five wide column {{if device.isMobile 'center' 'right'}} aligned right floated">
+      <span>{{t 'Show'}}</span>
+      {{#ui-dropdown class='selection'}}
```
Comment (niranjan94) [inline app/templates/events/view/tickets/access.hbs]: Like I said in the channel, remove this

File: app/templates/events/view/tickets/access.hbs (hunk 3 above)
Comment (niranjan94) [inline app/templates/events/view/tickets/access.hbs]: This is still not done @sumedh123 . Pass in data via an array from the route

File: app/templates/events/view/tickets/access.hbs [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+          <th>{{t 'Accessible Ticket'}}</th>
+          <th>{{t 'Validity'}}</th>
+          <th>{{t 'Status'}}</th>
+          <th>{{t 'Action'}}</th>
+        </tr>
+      </thead>
+      <tbody>
+        {{#each model as |entry|}}
+          <tr>
+            <td>{{entry.number}}</td>
```
Comment (niranjan94) [inline app/templates/events/view/tickets/access.hbs]: Serial Numbering should be something you do here. Won't come from the API

File: app/routes/events/view/tickets/access.js [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+export default Route.extend({
+  titleToken() {
+    return this.i18n.t('Access');
+  },
+
+  model() {
+    return [
+      {
+        'number'            : 0,
+        'access_code'       : '#386',
```
Comment (niranjan94) [inline app/routes/events/view/tickets/access.js]: use camelCase for all javascript object param names

File: app/templates/events/view/tickets/access.hbs [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+            <td>{{entry.access_code}}</td>
+            <td>{{entry.access_code_url}}</td>
+            <td>{{entry.accessible_ticket}}</td>
+            <td>{{entry.validity}}</td>
+            <td>{{entry.status}}</td>
+            <td>{{entry.action}}</td>
```
Comment (niranjan94) [inline app/templates/events/view/tickets/access.hbs]: Action should have buttons. Please see what is there in the current UI and don't make assumptions.

File: app/templates/events/view/tickets/access.hbs [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+      </thead>
+      <tbody>
+        {{#each model as |entry|}}
+          <tr>
+            <td>{{entry.number}}</td>
+            <td>{{entry.access_code}}</td>
+            <td>{{entry.access_code_url}}</td>
+            <td>{{entry.accessible_ticket}}</td>
+            <td>{{entry.validity}}</td>
+            <td>{{entry.status}}</td>
```
Comment (niranjan94) [inline app/templates/events/view/tickets/access.hbs]: Status should be formatted as a label with proper colors. Again, see the current UI

File: app/router.js [hunk 18]
```diff
@@ -50,6 +50,7 @@ router.map(function() {
         this.route('orders');
         this.route('attendees');
         this.route('add-order');
+        this.route('access');
```
Comment (niranjan94) [inline app/router.js]: call the route `access-codes`

**Comment by sumedh123** [comment]
@niranjan94 Please review. https://oefrontend.herokuapp.com

**Comment by niranjan94** [comment]
@sumedh123 none of the changes I have request seem to be done yet

**Comment by sumedh123** [comment]
@niranjan94 Sorry I did changes in my other PR and commented on this one by mistake

**Comment by mariobehling** [comment]
@sumedh123 Please improve on the quality of your commits. I see that there is lots of feedback required from other devs in this commit and it is still not merged. We need to get this project ready. The reason why so many GSoC students are working on this project is to get a result as fast as possible. Please help us to achieve this goal.

**Comment by sumedh123** [comment]
@mariobehling Sure. I will try my best to do this.

**Comment by sumedh123** [comment]
@niranjan94 Please review

File: app/templates/events/view/tickets/access.hbs [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+          <th>{{t 'Accessible Ticket'}}</th>
+          <th>{{t 'Validity'}}</th>
+          <th>{{t 'Status'}}</th>
+          <th>{{t 'Action'}}</th>
+        </tr>
+      </thead>
+      <tbody>
+        {{#each model as |entry index|}}
+          <tr>
+            <td>{{index}}</td>
```
Comment (niranjan94) [inline app/templates/events/view/tickets/access.hbs]: Index would be starting from zero. Use the increment helper for displaying this correctly. Please pay more attention to such detail

File: app/routes/events/view/tickets/access.js [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+  },
+
+  model() {
+    return [
+      {
+        'accessCode'       : '#386',
+        'accessCodeUrl'    : 'https://github.com',
+        'accessibleTicket' : 'Early bird',
+        'validity'         : '3 months',
+        'status'           : true
```
Comment (niranjan94) [inline app/routes/events/view/tickets/access.js]: What is this status ? just a status `true` doesn't really make any sense does it ?.
Should be called `isActive`

**Comment by sumedh123** [comment]
@niranjan94 Please review

**Comment by geekyd** [comment]
@sumedh123 Can you please provide deployment link?

**Comment by sumedh123** [comment]
@geekyd https://oefrontend2017.herokuapp.com

**Comment by geekyd** [comment]
@sumedh123 Please check why travis is failing.

File: app/templates/events/view/tickets/access.hbs [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+            <td>{{entry.accessCode}}</td>
+            <td>{{entry.accessCodeUrl}}</td>
+            <td>{{entry.accessibleTicket}}</td>
+            <td>{{entry.validity}}</td>
+            {{#if entry.isActive}}
+              <td><div class="ui label green center aligned">{{t 'Active'}}</div></td>
+            {{else}}
+              <td><div class="ui label red center aligned">{{t 'Inactive'}}</div></td>
+            {{/if}}
+            <td>
```
Comment (geekyd) [inline app/templates/events/view/tickets/access.hbs]: Please use buttons similar to ones used in `tickets/orders`

File: app/helpers/increment.js [hunk 22]
```diff
@@ -0,0 +1,9 @@
+import Ember from 'ember';
```
Comment (geekyd) [inline app/helpers/increment.js]: Please remove this use pre-existing helper. Check this https://github.com/DockYard/ember-composable-helpers#inc

**Comment by geekyd** [comment]
@sumedh123 Please remove the helper completely & use pre-existing helper. Thanks

**Comment by sumedh123** [comment]
@geekyd Check now. The heroku link is same.
https://oefrontend2017.herokuapp.com

File: app/templates/events/view/tickets/access.hbs [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+            <td>{{entry.accessCodeUrl}}</td>
+            <td>{{entry.accessibleTicket}}</td>
+            <td>{{entry.validity}}</td>
+            {{#if entry.isActive}}
+              <td><div class="ui basic mini green label center aligned">{{t 'Active'}}</div></td>
+            {{else}}
+              <td><div class="ui basic mini red label center aligned">{{t 'Inactive'}}</div></td>
+            {{/if}}
+            <td>
+              <button class="ui icon button">
```
Comment (geekyd) [inline app/templates/events/view/tickets/access.hbs]: This is still not done please use the buttons layout used in `tickets/orders`

Comment (sumedh123) [inline app/templates/events/view/tickets/access.hbs]: Done.

File: app/templates/events/view/tickets/access.hbs [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+            <td>{{entry.accessCode}}</td>
+            <td>{{entry.accessCodeUrl}}</td>
+            <td>{{entry.accessibleTicket}}</td>
+            <td>{{entry.validity}}</td>
+            {{#if entry.isActive}}
+              <td><div class="ui basic mini green label center aligned">{{t 'Active'}}</div></td>
+            {{else}}
+              <td><div class="ui basic mini red label center aligned">{{t 'Inactive'}}</div></td>
+            {{/if}}
+            <td class="center aligned">
```
Comment (geekyd) [inline app/templates/events/view/tickets/access.hbs]: Please remove `center aligned` from here.

File: app/templates/events/view/tickets/access.hbs [hunk 25]
```diff
@@ -0,0 +1,62 @@
...
+    <table class="ui very basic table">
+      <thead>
+        <tr>
+          <th>{{t '#'}}</th>
+          <th>{{t 'Access Code'}}</th>
+          <th>{{t 'Access Code URL'}}</th>
+          <th>{{t 'Accessible Ticket'}}</th>
+          <th>{{t 'Validity'}}</th>
+          <th>{{t 'Status'}}</th>
+          <th>{{t 'Action'}}</th>
```
Comment (geekyd) [inline app/templates/events/view/tickets/access.hbs]: Add `center aligned` also

Comment (sumedh123) [inline app/templates/events/view/tickets/access.hbs]: Then we need to center align data part also. i.e rows

Comment (geekyd) [inline app/templates/events/view/tickets/access.hbs]: No need to do this.

**Comment by sumedh123** [comment]
@geekyd @niranjan94 Please merge this one.

**Pull request merged**